### DISCUSSION TITLE
Fragmentation efficiency  - 232738 members ingested - 22574 processed

### DISCUSSION THREAD

**Discussion by xdxxxdx** [Informatievlaanderen/VSDS-LDESServer4J#876] [General]
Hello.

I am using the low water quality observations data set for testing 1.4.0 snapshot
Full setup please see: https://github.com/xdxxxdx/ODALA_demonstrator

**232738 water quality members are ingested.**
![image](https://github.com/Informatievlaanderen/VSDS-LDESServer4J/assets/15314807/54f1a697-5038-486d-bb6e-8dfad0a4d436)

**In the real-time: 22574 members are processed for the fragmentations.**
![image](https://github.com/Informatievlaanderen/VSDS-LDESServer4J/assets/15314807/ce611a19-a2a2-4be7-84d9-40da464020fb)
The gap is quite big for me, but don't know if there is an efficiency matrix set for evaluating the project.
Therefore, I make this case for future discussion.

Thanks

**Comment by Tomvbe** [comment]
Hi @xdxxxdx,

Decoupling the ingest from the fragmentation is intended behavior. We persist these in the database so a large gap won't cause memory issues and we will get eventual consistency.

However I do agree that the gap is quite big in your case and took the liberty to look into your ldes-server.yml. You did not activate auto-index-creation. I assume you aren't managing your indices manually either. This property has a very big impact on the performance of the fragmentation. Can you try adding the following property to your yml?

```
spring:
  data:
    mongodb:
      auto-index-creation: true
```

**Reply by xdxxxdx** [reply]
Thank @Tomvbe , will have a look, and give you a feedback.

**Reply by xdxxxdx** [reply]
Works perfectly for 562777 ldes members. thanks @Tomvbe
